src/sentry/tempest/tasks.py

```diff
@@ -0,0 +1,178 @@
+import logging
+
+from sentry import http
+from sentry.models.projectkey import ProjectKey, UseCase
+from sentry.tasks.base import instrumented_task
+from sentry.tempest.models import TempestCredentials
+from sentry.utils import json
+
+TEMPEST_URL = ""  # FIXME: Set this value once we have it
+POLL_LIMIT = 348  # 348 every 5 min ~ 100k a day
+
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.tempest.tasks.poll_tempest",
+    queue="tempest",
+)
+def poll_tempest(**kwargs):
+    # FIXME: Once we have more traffic this needs to be done smarter.
+    for credentials in TempestCredentials.objects.all():
+        if credentials.latest_fetched_item_id is None:
+            fetch_latest_item_id.delay(credentials.id)
+        else:
+            poll_tempest_crashes.delay(credentials.id)
+
+
+@instrumented_task(
+    name="sentry.tempest.tasks.fetch_latest_item_id",
+    queue="tempest",
+)
+def fetch_latest_item_id(credentials_id: int) -> None:
+    # FIXME: Try catch this later
+    credentials = TempestCredentials.objects.select_related("project").get(id=credentials_id)
+    project_id = credentials.project.id
+    org_id = credentials.project.organization_id  # this should work?
+    client_id = credentials.client_id
+
+    try:
+        response_text = fetch_latest_id_from_tempest(
+            org_id=org_id,
+            project_id=project_id,
+            client_id=client_id,
+            client_secret=credentials.client_secret,
+        )
+
+        if response_text.isdigit():  # If response is a number the fetch was a success
+            credentials.latest_fetched_item_id = response_text
+            credentials.save(update_fields=["latest_fetched_item_id"])
+            return
+
+        # Things the user ought to know about
+        if response_text.startswith("Invalid credentials"):
+            credentials.message = "Seems like the provided credentials are invalid"
+            credentials.save(update_fields=["message"])
+            return
+        if response_text.startswith("IP address not allow-listed"):
+            credentials.message = "Seems like our IP is not allow-listed"
+            credentials.save(update_fields=["message"])
+            return
+
+        logger.info(
+            "Fetching the latest item id failed.",
+            extra={"org_id": org_id, "project_id": project_id, "client_id": client_id},
+        )
+    except Exception:
+        logger.info(
+            "Fetching the latest item id failed.",
+            extra={"org_id": org_id, "project_id": project_id, "client_id": client_id},
+        )
+
+
+@instrumented_task(
+    name="sentry.tempest.tasks.poll_tempest",
+    queue="tempest",
```

Comment:
Similar as above, we should define a few more things when instrumenting the task: https://github.com/getsentry/sentry/pull/82454/files#r1895521917